packages/jest-environment-jsdom/src/index.ts

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import {Script} from 'vm';
+import {Global, Config} from '@jest/types';
+import mock, {ModuleMocker} from 'jest-mock';
+import {installCommonGlobals} from 'jest-util';
+import {JestFakeTimers as FakeTimers} from '@jest/fake-timers';
+import {JestEnvironment, EnvironmentContext} from '@jest/environment';
+import {JSDOM, VirtualConsole} from 'jsdom';
+
+// The `Window` interface does not have an `Error.stackTraceLimit` property, but
+// `JSDOMEnvironment` assumes it is there.
+interface Win extends Window {
+  Error: {
+    stackTraceLimit: number;
+  };
+}
+
+function isWin(globals: Win | Global.Global): globals is Win {
+  return (globals as Win).document !== undefined;
+}
+
+class JSDOMEnvironment implements JestEnvironment {
+  dom: JSDOM | null;
+  fakeTimers: FakeTimers<number> | null;
+  // @ts-ignore
+  global: Global.Global | Win | null;
+  errorEventListener: ((event: Event & {error: unknown}) => void) | null;
+  moduleMocker: ModuleMocker | null;
+
+  constructor(config: Config.ProjectConfig, options: EnvironmentContext = {}) {
+    this.dom = new JSDOM('<!DOCTYPE html>', {
+      pretendToBeVisual: true,
+      runScripts: 'dangerously',
+      url: config.testURL,
+      virtualConsole: new VirtualConsole().sendTo(options.console || console),
+      ...config.testEnvironmentOptions,
+    });
+
+    // `defaultView` returns a `Window` type, which is missing the
+    // `Error.stackTraceLimit` property. See the `Win` interface above.
+    const global = (this.global = this.dom.window.document
+      .defaultView as Win | null);
```

Comment:
just `as Win`, not `null`

Reply:
Same as above, `defaultView` is `Window | null` so it made sense to replicate that in the type assertion. Happy to change it though, just let me know.